plutus-core/plutus-core/test/Pretty/Golden/Readable/Examples/InterList/InterCons.golden

```diff
@@ -1,39 +1,47 @@
-/\a b ->
-  \(x : a)
-   (y : b)
-   (xs :
-      (\a b ->
+/\a#0 b#1 ->
+  \(x#2 : a#0)
+   (y#3 : b#1)
+   (xs#4 :
+      (\a#0 b#1 ->
          ifix
-           (\(rec :: ((* -> * -> *) -> *) -> *) (spine :: (* -> * -> *) -> *) ->
-              spine
-                ((\(interlist :: * -> * -> *) a b ->
-                    all r. r -> (a -> b -> interlist b a -> r) -> r)
-                   (\a b -> rec (\(dat :: * -> * -> *) -> dat a b))))
-           (\(dat :: * -> * -> *) -> dat a b))
-        b
-        a) ->
+           (\(rec#6 :: ((* -> * -> *) -> *) -> *)
+             (spine#7 :: (* -> * -> *) -> *) ->
+              spine#7
+                ((\(interlist#2 :: * -> * -> *) a#0 b#1 ->
+                    all r#3.
+                      r#3 -> (a#0 -> b#1 -> interlist#2 b#1 a#0 -> r#3) -> r#3)
```

Comment:
I feel like `#` is too noisy, I think `` ` `` would be perfect here, compare this

```
r#3 -> (a#0 -> b#1 -> interlist#2 b#1 a#0 -> r#3) -> r#3)
```

and this

```
r`3 -> (a`0 -> b`1 -> interlist`2 b`1 a`0 -> r`3) -> r`3)
```

Now we of course already use `` ` `` for quoted identifiers, but we could use something else, I think `@` might work, given that it's basic Haddock syntax and GHC uses the symbol for type applications which we render differently, so we won't clash with GHC here (still need escaping for non-GHC-produced code, but that's a given).

Reply:
Indeed, `` ` `` is lighter which could be more aesthetically pleasing, depending on your taste. I wouldn't mind us using `` ` ``, however these reasons make me think its not the best option:
1. More often than not back-ticks are used in pairs; When singular back-ticks happen to match they can hurt readability. Consider ``expected`0 risk`100`` -- it reads as if `0 risk` is spliced inside `expected100`
2. As you've mentioned: we already use back-ticks for escaping.

Between `` ` `` and `@` I chose the latter.

Lets also consider these characters:
```
ghci> import PlutusCore.Name.Unique
ghci> isIdentifierChar '~'
False
ghci> isIdentifierChar '!'
False
ghci> isIdentifierChar '-'
False
ghci> isIdentifierChar '^'
False
```

Reply:
Demo:
```
r-3 -> (a-0 -> b-1 -> interlist-2 b-1 a-0 -> r-3) -> r-3)
```
```
r^3 -> (a^0 -> b^1 -> interlist^2 b^1 a^0 -> r^3) -> r^3)
```
```
r~3 -> (a~0 -> b~1 -> interlist~2 b~1 a~0 -> r~3) -> r~3)
```
```
r!3 -> (a!0 -> b!1 -> interlist!2 b!1 a!0 -> r!3) -> r!3)
```
```
r|3 -> (a|0 -> b|1 -> interlist|2 b|1 a|0 -> r|3) -> r|3)
```

Reply:
Personally, I like `!` most of all above.

Reply:
> More often than not back-ticks are used in pairs; When singular back-ticks happen to match they can hurt readability. Consider ``expected`0 risk`100`` -- it reads as if `0 risk` is spliced inside `expected100`

Yeah, good point.

I just took a long look at the demo. I'm OK with any of those, but `-` seems very nice as well as `!` indeed (except I'd perhaps prefer the former over the latter).

Reply:
👍🏼 I went with `-` for `Unique` indexes and `!` for De Bruijn (instead of `#` for all)